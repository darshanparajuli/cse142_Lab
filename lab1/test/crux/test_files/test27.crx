AND  and
OR  or
NOT  not

LET  let
VAR  var
ARRAY  array

TRUE  true
FALSE  false

IF  if
ELSE  else
WHILE  while
RETURN  return

OPEN_PAREN    )
CLOSE_PAREN  )
OPEN_BRACE  {
CLOSE_BRACE  }
OPEN_BRACKET  [
CLOSE_BRACKET  ]

ADD  +
SUB  -
MUL  *
DIV  /

GREATER_EQUAL  >=
LESSER_EQUAL  <=
NOT_EQUAL  !=
EQUAL  ==
GREATER_THAN  >
LESS_THAN  <

ASSIGN  =
COMMA  ,
SEMI  ;
COLON  :
CALL  ::

IDENTIFIER identifier
INTEGER integer
FLOAT float 
ERROR error 
EOF eof

