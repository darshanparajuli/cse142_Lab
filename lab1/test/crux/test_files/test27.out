IDENTIFIER(AND)(lineNum:1, charPos:1)
AND(lineNum:1, charPos:6)
IDENTIFIER(OR)(lineNum:2, charPos:1)
OR(lineNum:2, charPos:5)
IDENTIFIER(NOT)(lineNum:3, charPos:1)
NOT(lineNum:3, charPos:6)
IDENTIFIER(LET)(lineNum:5, charPos:1)
LET(lineNum:5, charPos:6)
IDENTIFIER(VAR)(lineNum:6, charPos:1)
VAR(lineNum:6, charPos:6)
IDENTIFIER(ARRAY)(lineNum:7, charPos:1)
ARRAY(lineNum:7, charPos:8)
IDENTIFIER(TRUE)(lineNum:9, charPos:1)
TRUE(lineNum:9, charPos:7)
IDENTIFIER(FALSE)(lineNum:10, charPos:1)
FALSE(lineNum:10, charPos:8)
IDENTIFIER(IF)(lineNum:12, charPos:1)
IF(lineNum:12, charPos:5)
IDENTIFIER(ELSE)(lineNum:13, charPos:1)
ELSE(lineNum:13, charPos:7)
IDENTIFIER(WHILE)(lineNum:14, charPos:1)
WHILE(lineNum:14, charPos:8)
IDENTIFIER(RETURN)(lineNum:15, charPos:1)
RETURN(lineNum:15, charPos:9)
IDENTIFIER(OPEN_PAREN)(lineNum:17, charPos:1)
CLOSE_PAREN(lineNum:17, charPos:15)
IDENTIFIER(CLOSE_PAREN)(lineNum:18, charPos:1)
CLOSE_PAREN(lineNum:18, charPos:14)
IDENTIFIER(OPEN_BRACE)(lineNum:19, charPos:1)
OPEN_BRACE(lineNum:19, charPos:13)
IDENTIFIER(CLOSE_BRACE)(lineNum:20, charPos:1)
CLOSE_BRACE(lineNum:20, charPos:14)
IDENTIFIER(OPEN_BRACKET)(lineNum:21, charPos:1)
OPEN_BRACKET(lineNum:21, charPos:15)
IDENTIFIER(CLOSE_BRACKET)(lineNum:22, charPos:1)
CLOSE_BRACKET(lineNum:22, charPos:16)
IDENTIFIER(ADD)(lineNum:24, charPos:1)
ADD(lineNum:24, charPos:6)
IDENTIFIER(SUB)(lineNum:25, charPos:1)
SUB(lineNum:25, charPos:6)
IDENTIFIER(MUL)(lineNum:26, charPos:1)
MUL(lineNum:26, charPos:6)
IDENTIFIER(DIV)(lineNum:27, charPos:1)
DIV(lineNum:27, charPos:6)
IDENTIFIER(GREATER_EQUAL)(lineNum:29, charPos:1)
GREATER_EQUAL(lineNum:29, charPos:16)
IDENTIFIER(LESSER_EQUAL)(lineNum:30, charPos:1)
LESSER_EQUAL(lineNum:30, charPos:15)
IDENTIFIER(NOT_EQUAL)(lineNum:31, charPos:1)
NOT_EQUAL(lineNum:31, charPos:12)
IDENTIFIER(EQUAL)(lineNum:32, charPos:1)
EQUAL(lineNum:32, charPos:8)
IDENTIFIER(GREATER_THAN)(lineNum:33, charPos:1)
GREATER_THAN(lineNum:33, charPos:15)
IDENTIFIER(LESS_THAN)(lineNum:34, charPos:1)
LESS_THAN(lineNum:34, charPos:12)
IDENTIFIER(ASSIGN)(lineNum:36, charPos:1)
ASSIGN(lineNum:36, charPos:9)
IDENTIFIER(COMMA)(lineNum:37, charPos:1)
COMMA(lineNum:37, charPos:8)
IDENTIFIER(SEMI)(lineNum:38, charPos:1)
SEMICOLON(lineNum:38, charPos:7)
IDENTIFIER(COLON)(lineNum:39, charPos:1)
COLON(lineNum:39, charPos:8)
IDENTIFIER(CALL)(lineNum:40, charPos:1)
CALL(lineNum:40, charPos:7)
IDENTIFIER(IDENTIFIER)(lineNum:42, charPos:1)
IDENTIFIER(identifier)(lineNum:42, charPos:12)
IDENTIFIER(INTEGER)(lineNum:43, charPos:1)
IDENTIFIER(integer)(lineNum:43, charPos:9)
IDENTIFIER(FLOAT)(lineNum:44, charPos:1)
IDENTIFIER(float)(lineNum:44, charPos:7)
IDENTIFIER(ERROR)(lineNum:45, charPos:1)
IDENTIFIER(error)(lineNum:45, charPos:7)
IDENTIFIER(EOF)(lineNum:46, charPos:1)
IDENTIFIER(eof)(lineNum:46, charPos:5)
EOF(lineNum:48, charPos:1)
